File: Shape.java
import java.util.*;
import java.lang.*;
import java.io.*;
import java.util.ArrayList.*;

abstract class Shape
{

    double l,b;
    
    double area;
    abstract public void printarea();
    abstract public void input();
    abstract public double calarea();
}
class Rectangle extends Shape
{
    
    
    public void printarea()
	    {
	        
	        
            
            
            System.out.printf("\nRectangle: length is %s breadth is %s area is %s ",l,b,area);
            
            
        }
    
    public void input()
    {
        Scanner sc = new Scanner(System.in);
        System.out.print("enter the length ");
        l = sc.nextInt();
        System.out.print("enter the breadth ");
        b = sc.nextInt();
        
    }
    public double calarea()
    {
        area = l *b ;
        return area;
        
    }
    
}



class Square extends Shape
{
    public void printarea()
    {
        
        System.out.printf("\nsquare:  side is %s area is %s ",l,area);
        
    }
    
    public void input()
    {
        Scanner sc = new Scanner(System.in);
        System.out.print("enter the side ");
        l = sc.nextInt();
        
    }
    public double calarea()
    {
        area=l*l;
        return area;
    }
        
}



class Circle extends Shape
{
    public void printarea()
    {
        
        System.out.printf("\nCircle: radius is %s area is %s",l,area);
        
    }
    
    public void input()
    {
        Scanner sc = new Scanner(System.in);
        System.out.print("enter the radius ");
        l=sc.nextInt();
    }
    public double calarea()
    {
       double pi = 3.14;
       area=pi*l*l; 
       return area;
    }
    
}


class Triangle extends Shape
{
    public void printarea()
    {
        
        System.out.printf("\ntriangle: base is %s heaight is %s area is %s ",l,b,area);
        
        
    }
    
    public void input()
    {
        Scanner sc = new Scanner(System.in);
        System.out.print("enter the base ");
        l=sc.nextInt();
        System.out.print("enter the height ");
        b=sc.nextInt();
    }
    public double calarea()
    {
        area=0.5*l*b;
        return area;
    }
}



class Parallelogram extends Shape
{
    public void printarea()
    {
       
        System.out.printf("\nParallelogram: base is %s height is %s area is %s ",l,b,area);
        
    }
    
    public void input()
    {
        Scanner sc = new Scanner(System.in);
        System.out.print("enter the base ");
        l=sc.nextInt();
        System.out.print("enter the height ");
        b=sc.nextInt();
    }
    public double calarea()
    {
         area=l*b;
         return area;
    }
    
}



class Main
{
    
    public static void main(String[] args)
	{
	    
        int i;
        int j;
        
        double total=0;
        Shape s[] = new Shape[5];
        
        
        
        
        
        Scanner sc = new Scanner(System.in);
        
        System.out.print("enter total shapes ");
        int n = sc.nextInt();
        
      for(i=0;i<n;i++)
        {
            System.out.println("enter shape ");
            String shape = sc.next();
            if(shape.equals("rectangle"))
            {
                
                
                s[i]=new Rectangle();
                
                
                
                s[i].input();
                total=total+s[i].calarea();
                
                
                
                
            }
            else if(shape.equals("square"))
            {
                s[i]=new Square();
                s[i].input();
                total=total+s[i].calarea();
                
                
                
            }
            else if(shape.equals("circle"))
            {
                s[i]=new Circle();
                s[i].input();
                total=total+s[i].calarea();
            
                
            }
            else if(shape.equals("triangle"))
            {
                s[i]=new Triangle();
                s[i].input();
                total=total+s[i].calarea();
                
                
            }
            else if(shape.equals("parallelogram"))
            {
                s[i]=new Parallelogram();
                s[i].input();
                total=total+s[i].calarea();
                
               
                
            }
            
            else
            {
              System.out.println("sorry not found");  
            }
            
        }
        for(j=0;j<n;j++)
        {
            s[j].printarea();
        }
        System.out.println("\nthe total area is: " + total);
        
	}
}
            

File: Shape.py

from abc import ABC,abstractmethod

class Shape(ABC):
    
    global l,b,area
    
    
    
    
    
    @abstractmethod
    def inputuser(self):
        pass
    @abstractmethod
    def calarea(self):
        pass
    @abstractmethod
    def printarea(self):
        pass


class Rectangle(Shape):
    def inputuser(self):
        self.l=float(input("enter the length "))
        self.b=float(input("enter the breadth "))
    
    def printarea(self):
        
        print("rectangle: side is" +str(self.l)+ "breadth is " + str(self.b)+ "area is" + str(self.area))
    
    def calarea(self):
        self.area=self.l*self.b
        return self.area
    
        

class Square(Shape):

    def inputuser(self):
        self.l=float(input("enter the side "))
        
    
    def printarea(self):
        
        print("square: side is" +str(self.l)+ "area is" + str(self.area))
        
        
    def calarea(self):
        self.area=self.l*self.l
        return self.area
        
        
class Circle(Shape):

    def inputuser(self):
        self.l=float(input("enter the radius "))

    def printarea(self):
       
        print("circle: radius is" +str(self.l)+"area  is " + str(self.area))
        
    
    def calarea(self):
        pi=3.14
        self.area=pi*self.l*self.l
        return self.area
        
    
        
        
class Triangle(Shape):

    def inputuser(self):
        self.l=float(input("enter the base "))
        self.b=float(input("enter the height "))

    def printarea(self):
        
        print("triangle: base is " +str(self.l)+"height is "+str(self.b)+" area is " + str(self.area))
        return self.area
    
    def calarea(self):
        
        self.area=0.5*self.l*self.b
        return self.area
    
    
        
class Parallelogram(Shape):

    def inputuser(self):
        self.l=float(input("enter the base "))
        self.b=float(input("enter the height "))

    def printarea(self):
        
        print("Parallelogram: base is "+str(self.l)+"height is"+str(self.b)+"area is " + str(self.area))
         
        
    def calarea(self):
        self.area=self.l*self.b
        return self.area
    




Array=[]


total=0


n=int(input("enter total shapes "))
for i in range(0,n):
    shape=input("enter a shape ")
    if shape=="rectangle":
        r=Rectangle()
        r.inputuser()
        
        
        
        
        total=total+r.calarea()
        
        Array.append(r)
    
    elif shape=="square":
        s=Square()
        s.inputuser()
        
        
        
        
        total=total+s.calarea()
        
        Array.append(s)
    
    elif shape=="circle":
        c=Circle()
        c.inputuser()
        
        
        
        
        total=total+c.calarea()
        
        Array.append(c)
    
    elif shape=="triangle":
        t=Triangle()
        t.inputuser()
        
        
        
        
        total=total+t.calarea()
        
        Array.append(t)
    
    elif shape=="parallelogram":
        p=Parallelogram()
        p.inputuser()
        
        
        
        
        total=total+p.calarea()
        
        Array.append(p)
    else:
        print("shape was not found")

for i in Array:
    i.printarea()




    

print("the total area is "+ str(total))
